src/domain/interfaces/validator.interface.ts

```diff
@@ -0,0 +1,5 @@
+export interface IValidator<T> {
+  validate(data: unknown): T;
+  // TODO: deprecate validateMany in favor of: validatePage(data: unknown): Page<T>
+  validateMany(data: unknown[]): T[];
```

Comment:
When you mention to replace `validateMany` with `validatePage` does it mean at this level on the interface?

`validateMany` seems to put push functionality implementation to some validators even if it's not required. Eg.: Validating `T[]` would be done not by the Validator of `T` but by the Validator of `[]` (ArrayValidator). This way we can decouple `T` from Collection types when it comes to validation.

Reply:
Agree, I simplified the interface to have just a `validate` function so the caller keeps the responsibility to iterate the validation targets if needed.